clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -669,4 +669,6 @@ def err_drv_cuda_offload_only_emit_bc : Error<
 def warn_drv_jmc_requires_debuginfo : Warning<
   "/JMC requires debug info. Use '/Zi', '/Z7' or other debug options; option ignored">,
   InGroup<OptionIgnored>;
+def warn_drv_fsycl_with_c_type : Warning<
+  "treating '%0' input as '%1' when -fsycl is used">, InGroup<SyclCompat>;
```

Comment:
I'm not sure if we need to mention `when -fsycl is used` here. @premanandrao WDYT?

Reply:
I think there is some value is saying why the input type was changed.  Suggest quotes around -fsycl though:
```suggestion
  "treating '%0' input as '%1' when '-fsycl' is used">, InGroup<SyclCompat>;
```

I see this existing driver diagnostic as something that has a similar format:
```
 warn_drv_treating_input_as_cxx : Warning<
  "treating '%0' input as '%1' when in C++ mode, this behavior is deprecated">,
  InGroup<Deprecated>;

```